### PR TITLE
New header layouts

### PR DESCRIPTION
**Pull request by madalingorbanescu**

### CODE REVIEW

File: packages/block-library/src/blocks/header/editor-styles.scss [hunk 1]
```diff
@@ -13,3 +17,72 @@
 .wp-block-novablocks-navigation ul li[class][class] {
 	margin-top: 0;
 }
+
+.site-header-logo-left {
+  
+  .site-header__row--primary {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/editor-styles.scss:23]: While row inner blocks can be moved around classes like `.site-header__row--primary` `.site-header__row--secondary` or `.site-header__row--logo` seem irrelevant. I would try to obtain some less specific styles.

File: packages/block-library/src/blocks/header/variations.js [hunk 2]
```diff
@@ -2,37 +2,127 @@ import * as icons from './icons';
...
+          } ]
+        ]
+      ]
+    ],
+    scope: [ 'block' ],
+  },
+  {
+    name: 'logo-center-three-rows',
+    title: __('Three Rows Header with centered logo and three menus.', '__plugin_txtd'),
+    icon: icons.felt,
```
Comment (razwan) [inline packages/block-library/src/blocks/header/variations.js:87]: `logo-center-three-rows` better explains the intention of this variation than `felt`. Therefore, using theme names for accessing the icons for these variations has the only advantage of being short. This also applies for all the other variations.

File: packages/block-library/src/blocks/header/style.scss [hunk 3]
```diff
@@ -1,61 +1,230 @@
...
-		}
-	}
-
-	.c-menu-toggle__checkbox,
-	.c-menu-toggle {
-		display: none;
-	}
+  // Nova Blocks hacks
+  .site-header,
+  .site-header-sticky {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/style.scss:6]: Using only one classname for these styles would save use some bytes and the world some CO2

File: packages/block-library/src/blocks/header/edit.js [hunk 4]
```diff
@@ -98,6 +109,10 @@ class Edit extends Component {
 		}
 	}
 
+	componentDidUpdate() {
+    addSocialMenuClass.call(this);
```
Comment (razwan) [inline packages/block-library/src/blocks/header/edit.js:113]: `addSocialMenuClass` doesn't have any usages of `this` rendering the usage of the `call` method useless.
`addSocialMenuClass()` should suffice.

Comment (razwan) [reply packages/block-library/src/blocks/header/edit.js:113]: This could be optimised because `componentDidUpdate` is triggered often. However, firstly the Edit component should be rewritten using hooks.

File: packages/block-library/src/blocks/header/utils.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+export const addSocialMenuClass = function() {
+  const menuItem = document.querySelectorAll( '.menu-item a' );
```
Comment (razwan) [inline packages/block-library/src/blocks/header/utils.js:2]: Looking only inside `.site-header__menu` would be a good optimisation

File: packages/block-library/src/blocks/header/style.scss [hunk 6]
```diff
@@ -1,61 +1,230 @@
...
+}
+
+@include above(lap) {
+  
+  .site-header__row {
+    
+    display: flex;
+    align-items: center;
+    
+    &--secondary {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/style.scss:113]: Styling could be done based on the block-type and whether or not it is the first / last block inside the header row. I suggest trying this approach and only after that styling specific cases

File: lib/extras.php [hunk 7]
```diff
@@ -189,6 +189,14 @@ function novablocks_get_header_attributes() {
 			'type'    => 'string',
 			'default' => 'logo-left',
 		),
+		'stickyRow' => array(
+			'type'    => 'string',
+			'default' => 'primary'
+		),
+		'shouldBeSticky' => array(
```
Comment (razwan) [inline lib/extras.php:196]: This attribute seems redundant. The should be a sticky header if any of the header rows are sticky.

File: packages/block-library/src/blocks/header/deprecated.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      if (headerShouldBeSticky) {
+        attributes.shouldBeSticky = headerShouldBeSticky;
+      }
+
+      return [
+        omit(attributes, 'stickyRow' ),
+        [
+          createBlock( 'novablocks/header-row', {
+              className: 'site-header__row site-header__row--primary',
+              headerRowType: 'primary',
```
Comment (razwan) [inline packages/block-library/src/blocks/header/deprecated.js:52]: ~~headerRowType: 'primary',~~
isPrimary: true,

**Review by razwan — CHANGES_REQUESTED** [review]

File: packages/block-library/src/blocks/header-row/index.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  description: __( 'Outputs header row markup.', '__plugin_txtd' ),
+  category: 'nova-blocks',
+  parent: 'novablocks/header',
+  attributes: {
+    isSticky: {
+      type: 'boolean',
+      default: false
+    },
+    headerRowType: {
+      type: 'string',
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply packages/block-library/src/blocks/header-row/index.js:26]: Yes, there is a reason. I am using ``headerRowType`` to also [generate the header rows for options](headerRowType). In this way, we will expose only the rows that are inside the header block.

[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply packages/block-library/src/blocks/header-row/index.js:26]: Yes, there is a reason. I am using headerRowType to [generate the options for Sticky Row.](https://github.com/pixelgrade/nova-blocks/blob/3147b6d5fba6a21d8526a233585c6f83a6f15a11/packages/block-library/src/blocks/header/inspector-controls.js#L31).

File: lib/extras.php [hunk 10]
```diff
@@ -189,6 +189,14 @@ function novablocks_get_header_attributes() {
 			'type'    => 'string',
 			'default' => 'logo-left',
 		),
+		'stickyRow' => array(
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply lib/extras.php:192]: Yes, one of them is unnecessary, but in this situation is the one on the header row block, because I was not using it.

File: packages/block-library/src/blocks/header/style.scss [hunk 11]
```diff
@@ -1,61 +1,156 @@
...
-		display: none;
-	}
+@include above(lap) {
+  
+  .site-header__row {
+    
+    display: flex;
+    align-items: center;
+    
+    > *:first-child:not(:last-child) {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/style.scss:112]: `*` is redundant here

File: packages/block-library/src/blocks/header/style.scss [hunk 12]
```diff
@@ -1,61 +1,156 @@
...
+      justify-content: flex-start;
+      flex-shrink: 1;
+      flex-basis: 50%;
+    }
+    
+    > *:only-child {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/style.scss:118]: `*` is redundant here

File: packages/block-library/src/blocks/header/style.scss [hunk 13]
```diff
@@ -1,61 +1,156 @@
...
+      flex-grow: 1;
+      
+      &,
+      .menu {
+        justify-content: center;
+  
+      }
+    }
+    
+    > *:not(:first-child):not(:last-child) {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/style.scss:129]: `*` is redundant here

File: packages/block-library/src/blocks/header/style.scss [hunk 14]
```diff
@@ -1,61 +1,156 @@
...
+      justify-content: center;
+    }
+    
+    > *:last-child:not(:first-child) {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/style.scss:133]: `*` is redundant here

File: packages/block-library/src/blocks/header/init.php [hunk 15]
```diff
@@ -18,39 +27,210 @@ function novablocks_render_header_block( $attributes, $content ) {
...
+		// Helper class for Header layout
+		// that is on only one row.
+		if ( $header_is_simple ) {
+			$classes[] = 'site-header--simple';
+		}
+
+		// We need that class to style header block,
+		// if the user didn't hit save yet.
+		if ( ! headerBlockUpdated() ) {
+			$classes[] = 'site-header--is-old';
```
Comment (razwan) [inline packages/block-library/src/blocks/header/init.php:56]: Isn't wrapping the content inside a header-row block a better option?

File: packages/block-library/src/blocks/header/init.php [hunk 16]
```diff
@@ -18,39 +27,210 @@ function novablocks_render_header_block( $attributes, $content ) {
...
+		// Get Sticky Row Block.
+		$stickyRowBlock = getStickyRowBlock();
+
+		// Get Primary Row Block to use it on hover if it's the case.
+		$primaryRowBlock = getPrimaryBlock();
+
+		// Helper class for Header layout
+		// that is on only one row.
+		if ( $header_is_simple ) {
+			$classes[] = 'site-header--simple';
```
Comment (razwan) [inline packages/block-library/src/blocks/header/init.php:50]: This might not be needed anymore

File: packages/block-library/src/blocks/header/style.scss [hunk 17]
```diff
@@ -1,61 +1,156 @@
...
+      display: flex;
+      flex-direction: column;
+      
+      .site-header__menu--primary {
+        order: -1;
+      }
+    }
+  }
+  
+  .site-header__row--primary {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/style.scss:100]: garbage?

File: packages/block-library/src/blocks/header/style.scss [hunk 18]
```diff
@@ -1,61 +1,156 @@
...
+    pointer-events: auto;
+  }
+}
 
-		.site-header--logo-right & {
-			grid-template-columns: 1fr auto;
-		}
+.site-header {
+  
+  &--is-sticky {
```
Comment (razwan) [inline packages/block-library/src/blocks/header/style.scss:76]: `.site-header--is-sticky` would be more clearer since `.site-header` isn't used at all. Also, `--sticky` would be a better modifier for the `site-header` class, while `is-sticky` would make more sense as a separate class.

`.site-header-sticky` and `.site-header--is-sticky` create some confusion at this point while they even have the same rules:
```css
    position: fixed;
    top: var(--admin-bar-height, 0);
```

I would look into renaming / reusing more classes to have classes with more precise scopes. I would keep `site-header` on all site-headers and maybe change the JS selector to something like `js-site-header`, and target the main site header with something like `site-header--main` or `site-header--primary` or `site-header--static`.

I would use the `site-header--sticky` class for making any of the two sticky, and then hide on mobile only the header with the class `site-header--secondary` or `site-header--fixed`. The classes I used below are only for demonstration purposes. The `site-header--default` class may be useless in the example below but I wanted just make a point about consistency in naming.

```html
<!-- main header -->
<div class="site-header site-header--default site-header--static [site-header--sticky] js-site-header">

<!-- sticky header -->
<div class="site-header site-header--default site-header--fixed site-header--sticky">

File: packages/block-library/src/blocks/header/init.php [hunk 19]
```diff
@@ -18,39 +27,210 @@ function novablocks_render_header_block( $attributes, $content ) {
...
+		$classes = array( 'site-header site-header--default alignfull' );
+
+		$classes[] = 'site-header--' . $attributes['layout'];
+
+		// Logo Center and Logo Left layout are considered as simple,
+		// because they are on only one row.
+		$header_is_simple = $attributes['layout'] === 'logo-left' || $attributes['layout'] === 'logo-center';
+
+		// Get the sticky row.
+		$sticky_row = getStickyRowBlockLabel();
```
Comment (razwan) [inline packages/block-library/src/blocks/header/init.php:39]: This looks like it isn't needed anymore

**Review by razwan — CHANGES_REQUESTED** [review]
